### PR TITLE
New web rtc

### PR DESCRIPTION
**Pull request by JoeyDarbyshire**
Completes #6

### CODE REVIEW

File: browser/webRTC/client.js [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+const USE_AUDIO = true;
+const USE_VIDEO = false;
+const DEFAULT_CHANNEL = 'some-global-channel-name';
+/** You should probably use a different stun server doing commercial stuff **/
+/** Also see: https://gist.github.com/zziuni/3741933 **/
+const ICE_SERVERS = [
+  { url: 'stun:stun.l.google.com:19302' }
+];
+
+// This will be out socket connection
```
Comment (parky22) [inline browser/webRTC/client.js]: our

File: browser/webRTC/client.js [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+  /* Ask user for permission to use the computers microphone and/or camera,
+  * attach it to an <audio> or <video> tag if they give us access. */
+  console.log('Requesting access to local audio / video inputs');
+  navigator.getUserMedia({ 'audio': USE_AUDIO, 'video': USE_VIDEO }, function (stream) {
+    console.log('Access granted to audio/video');
+    localMediaStream = stream;
+    const audioEl = document.getElementById('localAudio');
+    audioEl.muted = true;
+    audioEl.srcObject = stream;
+    joinChatChannel(DEFAULT_CHANNEL, { 'whatever-you-want-here': 'stuff' });
```
Comment (parky22) [inline browser/webRTC/client.js:36]: are we going to implement a chat channel eventually? If not, do we need this?

Comment (parky22) [reply browser/webRTC/client.js:36]: ah nevermind, saw your comment below on line 59.

File: browser/socket.js [hunk 3]
```diff
@@ -67,3 +69,18 @@ socket.on('removeUser', userId => {
...
   avatarToBeRemoved.parentNode.removeChild(avatarToBeRemoved); // Remove from DOM
 });
+
+// Adds a Peer to our DoM as their own Audio Element
+socket.on('addPeer', addPeerConn);
+
+// Removes Peer from DoM after they have disconnected or switched room
+socket.on('removePeer', removePeerConn);
+
+// Replys to an offer made by a new Peer
```
Comment (bethqiang) [inline browser/socket.js]: Replies :)

**Review by bethqiang — APPROVED** [review]

**Review by parky22 — APPROVED** [review]

**Review by bushidocodes — APPROVED** [review]

**Pull request merged**